ksql-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/Locator.java

```diff
@@ -28,16 +30,18 @@
 public interface Locator {
 
   /**
-   * Locate which KSQL node stores the supplied {@code key}.
+   * Locate which KSQL nodes store the supplied {@code key}.
    *
    * <p>Implementations are free to return {@link Optional#empty()} if the location is not known at
    * this time.
    *
    * @param key the required key.
-   * @return the owning node, if known.
+   * @return the list of nodes, that can potentially serve the key.
    */
-  Optional<KsqlNode> locate(Struct key);
-
+  List<KsqlNode> locate(
+      Struct key,
+      RoutingFilter routingFilters
```

Comment:
I would have had a List<RoutingFilter> passed in here and have the KsLocator make use of RoutingFilters. But okay with this as well, in the interest of moving forward

Reply:
Look at Andy's comment here: https://github.com/confluentinc/ksql/pull/4398#discussion_r373450260

Reply:
ha ha